cli/docker/container_manager.go

```diff
@@ -0,0 +1,545 @@
+//     it can't automatically use the docker config
+func (cm *ContainerManager) runContainer(
+	container models.Container,
+	options containerCreateOptions,
+	envs map[string]string,
+) (*RunningContainer, error) {
+	if cm.released {
+		return nil, fmt.Errorf("container manager was released already")
+	}
+
+	if err := cm.ensureNetwork(); err != nil {
+		return nil, fmt.Errorf("ensure bitrise docker network: %w", err)
+	}
+
+	cm.logger.Infof("ℹ️ Pulling docker image: %s", container.Image)
+	err := cm.pullImageWithRetry(container)
+	if err != nil {
+		return nil, fmt.Errorf("pull docker image: %w", err)
+	}
+	cm.logger.Infof("✅ Docker image pulled: %s", container.Image)
+
+	cm.logger.Infof("ℹ️ Creating docker container: %s", container.Image)
+	err = cm.createContainer(container, options, envs)
+	if err != nil {
+		return nil, fmt.Errorf("create docker container: %w", err)
+	}
+	cm.logger.Infof("✅ Docker container created: %s", container.Image)
+
+	cm.logger.Infof("ℹ️ Starting docker container: %s", container.Image)
+	runningContainer, err := cm.startContainer(options)
+	if err != nil {
+		return runningContainer, fmt.Errorf("start docker container: %w", err)
+	}
+	cm.logger.Infof("✅ Container (%s) is running (%s)", runningContainer.Name, runningContainer.ID)
+
+	return runningContainer, nil
+}
+
+func (cm *ContainerManager) startContainer(options containerCreateOptions) (*RunningContainer, error) {
+	// At this point the container has been created, but it's not running yet
+	// Even if we can't start it we need to return the container reference to make sure it will be cleaned up
+	runningContainer := &RunningContainer{
+		Name: options.name,
+	}
+
+	cm.logger.Infof("ℹ️ Running command: docker start %s", options.name)
+	out, err := command.New("docker", "start", options.name).RunAndReturnTrimmedCombinedOutput()
+	if err != nil {
+		cm.logger.Errorf(out)
+		return runningContainer, fmt.Errorf("start docker container (%s): %w", options.name, err)
+	}
+
+	// We need to get the container ID to be able to check the health
+	// This also serves as a validation that the container is running
+	result, err := cm.getRunningContainer(context.Background(), options.name)
+	if result != nil {
+		runningContainer.ID = result.ID
+	}
+	if err != nil {
+		return runningContainer, fmt.Errorf("container (%s) unable to start properly: %w", options.name, err)
+	}
+	return runningContainer, nil
+}
+
+func (cm *ContainerManager) createContainer(
+	container models.Container,
+	options containerCreateOptions,
+	envs map[string]string,
+) error {
+	dockerRunArgs := []string{"create",
+		"--platform", "linux/amd64",
+		"--network=bitrise",
+	}
+
+	for _, o := range options.volumes {
+		dockerRunArgs = append(dockerRunArgs, "-v", o)
+	}
+
+	for _, env := range container.Envs {
+		for name, value := range env {
+			resolvedValue := resolveEnvVariable(fmt.Sprintf("%s", value), envs)
```

Comment:
Why `fmt.Sprintf()`?

Reply:
Because the envman implementation or type is defining the value as `interface{}`, aka. legacy reasons.